src/redisearch_rs/inverted_index/tests/freqs_only.rs

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+use std::io::Cursor;
+
+use inverted_index::{
+    Decoder, DecoderResult, Delta, Encoder, RSIndexResult, freqs_only::FreqsOnly,
+};
+
+#[test]
+fn test_encode_freqs_only() {
+    // Test cases for the frequencies only encoder and decoder.
+    let tests = [
+        // (frequency, delta, expected encoding)
+        (0, 0, vec![0, 0, 0]),
+        (0, 1, vec![0, 1, 0]),
```

Comment:
I don't think we should expect frequency 0, but I don't see any harm in testing it